### PR TITLE
Add simple YAML test for access control cluster

### PR DESCRIPTION
**Pull request by mlepage-google**
#### Problem
Need cluster level tests

#### Change overview
Add simple tests for now. They will be improved soon.

Towards issue #10253

#### Testing
* Ran tests locally on Linux.
* CI

### CODE REVIEW

**Review by woody-apple — APPROVED** [review]
Fast tracking added tests

File: src/app/tests/suites/TestAccessControlCluster.yaml [hunk 1]
```diff
@@ -0,0 +1,167 @@
...
+                      Targets:
+                          [
+                              { Cluster: null, Endpoint: 2, DeviceType: null },
+                              { Cluster: 3, Endpoint: null, DeviceType: null },
+                              { Cluster: 4, Endpoint: 5, DeviceType: null },
+                          ],
+                  },
+              ]
+
+    - label: "Write"
```
Comment (bzbarsky-apple) [inline src/app/tests/suites/TestAccessControlCluster.yaml:117]: Having non-repeating labels would really help with understanding the test output.....

File: src/app/tests/suites/TestAccessControlCluster.yaml [hunk 2]
```diff
@@ -0,0 +1,167 @@
...
+      command: "writeAttribute"
+      attribute: "ACL"
+      arguments:
+          value: [
+                  {
+                      FabricIndex: 1,
```
Comment (bzbarsky-apple) [inline src/app/tests/suites/TestAccessControlCluster.yaml:123]: Please use `FabricIndex: 0` on the writes (since it gets ignored on the server anyway).

**Review by bzbarsky-apple — APPROVED** [review]

**Review by jmeg-sfy — APPROVED** [review]

File: src/app/tests/suites/TestAccessControlCluster.yaml (hunk 1 above)
Comment (mlepage-google) [reply src/app/tests/suites/TestAccessControlCluster.yaml]: They do get replaced with write1 write2 etc. I envision having a whole series of these, for testing various values. I'll be in this file again soon. But for now I changed the labels.

File: src/app/tests/suites/TestAccessControlCluster.yaml (hunk 2 above)
Comment (mlepage-google) [reply src/app/tests/suites/TestAccessControlCluster.yaml]: It's easier to copy/paste the value from write to read test steps if they are the same. But it's also easier to just do the recommendation for now, I'll be back in this file soon and will be restructuring anyways.

**Comment by mlepage-google** [comment]
This YAML exposed an issue with YAML handling/codegen on Darwin, so it won't pass without PR #15276

**Pull request merged**